podpac/core/interpolation/xarray_interpolator.py

```diff
@@ -0,0 +1,123 @@
+    """Xarray interpolation Interpolation
+
+    {nearest_neighbor_attributes}
+    """
+
+    dims_supported = ["lat", "lon", "alt", "time"]
+    methods_supported = [
+        "nearest",
+        "linear",
+        "bilinear",
+        "quadratic",
+        "cubic",
+        "zero",
+        "slinear",
+        "next",
+        "previous",
+        "splinef2d",
+    ]
+
+    # defined at instantiation
+    method = tl.Unicode(default_value="nearest")
+    fill_value = tl.Union([tl.Unicode(), tl.Float()], default_value=None, allow_none=True)
+    fill_nan = tl.Bool(False)
+
+    kwargs = tl.Dict({"bounds_error": False})
+
+    def __repr__(self):
+        rep = super(XarrayInterpolator, self).__repr__()
+        # rep += '\n\tspatial_tolerance: {}\n\ttime_tolerance: {}'.format(self.spatial_tolerance, self.time_tolerance)
+        return rep
+
+    @common_doc(COMMON_INTERPOLATOR_DOCS)
+    def can_interpolate(self, udims, source_coordinates, eval_coordinates):
+        """
+        {interpolator_interpolate}
+        """
+        udims_subset = self._filter_udims_supported(udims)
+
+        # confirm that udims are in both source and eval coordinates
+        if self._dim_in(udims_subset, source_coordinates, unstacked=True):
+            for d in source_coordinates.udims:  # Cannot handle stacked dimensions
+                if source_coordinates.is_stacked(d):
+                    return tuple()
+            return udims_subset
+        else:
+            return tuple()
+
+    @common_doc(COMMON_INTERPOLATOR_DOCS)
+    def interpolate(self, udims, source_coordinates, source_data, eval_coordinates, output_data):
+        """
+        {interpolator_interpolate}
+        """
+        indexers = []
+
+        coords = {}
+        nn_coords = {}
+        used_dims = set()
+
+        for d in udims:
+            if source_coordinates[d].size == 1:
+                # If the source only has a single coordinate, xarray will automatically throw an error asking for at least 2 coordinates
+                # So, we prevent this. Main problem is that this won't respect any tolerances.
+                new_dim = [dd for dd in eval_coordinates.dims if d in dd][0]
+                nn_coords[d] = xr.DataArray(
+                    eval_coordinates[d].coordinates,
+                    dims=[new_dim],
+                    coords=[eval_coordinates.xcoords[new_dim]],
+                )
+                continue
+            if not source_coordinates.is_stacked(d) and eval_coordinates.is_stacked(d):
+                new_dim = [dd for dd in eval_coordinates.dims if d in dd][0]
+                coords[d] = xr.DataArray(
+                    eval_coordinates[d].coordinates, dims=[new_dim], coords=[eval_coordinates.xcoords[new_dim]]
+                )
+
+            elif source_coordinates.is_stacked(d) and not eval_coordinates.is_stacked(d):
+                raise InterpolatorException("Xarray interpolator cannot handle multi-index (source is points).")
```

Comment:
What's going on here? The Xarray Interpolator can handle stacked source to stacked eval, but not stacked source to unstacked eval? Why is it raised as an exception instead checked in `can_interpolate`?

Reply:
Yep, that makes no sense. It's actually already handled in the `can_interpolate` function. This shouldn't be there. Removed (in a forthcoming commit).